### PR TITLE
Upgrade Go version to 1.23.1

### PR DESCRIPTION
**Pull request by wildum**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/grafana/alloy
 
-go 1.22.5
+go 1.23.1
```
Comment (mattdurham) [inline go.mod:3]: Do we need to also update pyroscope here? I know there was a collision between pyroscope pprof endpoint and go 1.23?

Comment (mattdurham) [reply go.mod:3]: Run the exe and it should trigger either immediately or hitting the pprof endpoint

Comment (aleks-p) [reply go.mod:3]: The CI caught it as well it seems. Bumping `github.com/grafana/pyroscope-go/godeltaprof` to 0.1.8 will likely fix it. Here is a related fix in Grafana: https://github.com/grafana/grafana/pull/92055

Comment (wildum) [reply go.mod:3]: thanks, I will do it

**Comment by mattdurham** [comment]
kwe may need to separate the build image from the executable version change.

**Comment by wildum** [comment]
yeah this is what I'm trying to do but boringcrypto is not happy

**Pull request closed without merging**